### ISSUE TITLE
`undefined` is appearing in various `.navItem` elements

### ISSUE THREAD

**Issue by ericscheid** [naturalcrit/homebrewery#2693]
### What happened?

On the userpage, brew-edit page, and brew-preview pages; some of the drop-down `.navItem` elements have `undefined` class attributes. Not all though.

- Userpage
  - Need Help? — `undefined`
  - Recent Brews — ok
  - [Account Name] — `undefined`

- Brew Edit 
  - Save — ok
  - Need Help? — `undefined`
  - Share — `undefined`
  - Recent Brews — ok
  - [Account Name] — `undefined`

- Brew Share
  - Source — `undefined`
  - Recent Brews — ok
  - [Account Name] — `undefined`

<img width="897" alt="image" src="https://user-images.githubusercontent.com/2605467/220864495-31296cfb-eb34-494b-a75a-a6fc9651cddc.png">

This is a code smell. Something has been done wrong but is not causing any breakage. Yet.

----

Additionally, there appear to be two different structures for coding drop-down menus.

- a parent div containing a child div for the menu title and an icon
  - most of these have a class of `navDropdownContainer` and `undefined`
  - the save menu is an outlier, with the parent div having a `save-menu` class instead of undefined
- a parent div containing just a text node and an icon e.g. Recent Brews
  - the Recent Brews menu further deviates by not having a class of `navDropdownContainer`, despite being a drop-down menu

## Proposal

- Reviw the code for the different drop-down menus and conform to a consistent form that doesn't emit `undefined`

**Comment by ericscheid** [comment]
The ideal, I guess, is how the save menu is coded:

```html
<div class="navDropdownContainer save-menu">
  <div class="navItem blue save" color="blue" icon="fas fa-save">Save Now<i class="fas fa-save"></i></div>
</div>
```

The parent div has `navDropdownContainer`, no `undefined`, and a name (`save-menu`). 
Inside, a child div for the menu title with the icon.

**Comment by Gazook89** [comment]
Perhaps we should utilize `a` and `Button` tags as well. Rather than just divs. 

I might work on this next, unifying the elements in that top bar.

**Comment by G-Ambatte** [comment]
Here's the offending piece of code:

![image](https://github.com/naturalcrit/homebrewery/assets/19826659/aaf30ef2-dc16-4460-9bd4-dcaadc45b58d)

In the event that no `className` is provided, `this.props.className` returns `undefined`.
It should be pretty easy to fix, by changing this:

```
<div className={`navDropdownContainer ${this.props.className}`}
```

to something like:

```
<div className={`navDropdownContainer ${this.props.className || ''}`}
```

It might be simpler to just change the `getDefaultProps` function to return a 'className: ''` default prop.

**Comment by 5e-Cleric** [comment]
Fixed in #3291, didn't even know this issue existed at the time.
